packages/openapi-spec-builder/src/__tests__/unit/openapi-spec-builder.unit.ts

```diff
@@ -0,0 +1,143 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/openapi-v3
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {createEmptyApiSpec, ParameterObject} from '@loopback/openapi-v3-types';
+import {expect} from '@loopback/testlab';
+import {anOpenApiSpec, anOperationSpec} from '../../openapi-spec-builder';
+
+describe('OpenAPI Spec Builder', () => {
+  describe('anOpenApiSpec', () => {
+    it('creates an empty spec', () => {
+      const spec = anOpenApiSpec().build();
+      expect(spec).to.eql(createEmptyApiSpec());
+    });
+
+    it('adds an extension', () => {
+      const spec = anOpenApiSpec()
+        .withExtension('x-loopback-version', '4.0')
+        .build();
+      expect(spec).to.containEql({'x-loopback-version': '4.0'});
+    });
+
+    it('adds an operation', () => {
+      const opSpec = anOperationSpec().build();
+      const spec = anOpenApiSpec()
+        .withOperation('get', '/users', opSpec)
+        .build();
+      expect(spec.paths).to.containEql({
+        '/users': {
+          get: opSpec,
+        },
+      });
+    });
+  });
+
+  describe('anOperationSpec', () => {
+    it('creates an empty spec', () => {
+      const spec = anOperationSpec().build();
+      expect(spec).to.eql({
+        responses: {'200': {description: 'An undocumented response body.'}},
+      });
+    });
+
+    it('adds an extension', () => {
+      const spec = anOperationSpec()
+        .withExtension('x-loopback-authentication', 'oAuth2')
+        .build();
+      expect(spec).to.containEql({'x-loopback-authentication': 'oAuth2'});
+    });
+
+    it('sets controller name', () => {
+      const spec = anOperationSpec()
+        .withControllerName('MyController')
+        .build();
+      expect(spec).to.containEql({'x-controller-name': 'MyController'});
+    });
+
+    it('sets operation name', () => {
+      const spec = anOperationSpec()
+        .withOperationName('greet')
+        .build();
+      expect(spec).to.containEql({'x-operation-name': 'greet'});
+    });
+
+    it('sets operationId', () => {
+      const spec = anOperationSpec()
+        .withOperationId('MyController.greet')
+        .build();
+      expect(spec).to.containEql({operationId: 'MyController.greet'});
+    });
+
+    it('sets operationId from controller/operation name', () => {
+      const spec = anOperationSpec()
+        .withControllerName('MyController')
+        .withOperationName('greet')
+        .build();
+      expect(spec).to.containEql({operationId: 'MyController$greet'});
```

Comment:
Depending on how we resolve our discussion above, this may need to be changed to

```suggestion
      expect(spec).to.containEql({operationId: 'MyController.greet'});
```